### PR TITLE
[Android] Add work unit buffer to preference

### LINKED ISSUE #2755: [Android] Add "Store at least X days of work" preference
On Android the "Store at least X days of work" and the "Store up to an additional X days of work" preferences would be nice.

### PR DESCRIPTION
**Pull request by adamradocz**
Fixes #2755

**Description of the Change**
Add WU buffer option to the Preference menu.

**Release Notes**
[Android] Add "Work unit buffer" option

### CODE REVIEW

**Comment by adamradocz** [comment]
@AenBleidd could you help me out? I can't find the function that links the Android preferences with the global_prefs_override.xml. How can I serialize it?

**Comment by adamradocz** [comment]
I think I got it.

**Comment by adamradocz** [comment]
Only the "Store at least N days of work" added, the "Store up to an additional N days of work" not.
Is that necessary?

**Comment by brevilo** [comment]
> Is that necessary?

Well, you requested it in the first place 😁 Since these are advanced options I personally don't care. Whatever gets implemented can be useful but there's no strict requirement. Bottom line: if you wan't to take it on it would certainly be appreciated and if you don't no one will blame you 👍

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/PrefsFragment.java [hunk 1]
```diff
@@ -407,7 +410,8 @@ public void onClick(View v) {
          	   } else if(item.ID == R.string.prefs_network_daily_xfer_limit_mb_header ||
          			   item.ID == R.string.battery_temperature_max_header ||
          			   item.ID == R.string.prefs_disk_min_free_gb_header ||
-         			   item.ID == R.string.prefs_disk_access_interval_header) {
+         			   item.ID == R.string.prefs_disk_access_interval_header ||
+					   item.ID == R.string.prefs_other_store_at_least_x_days_of_work_header) {
```
Comment (brevilo) [inline android/BOINC/app/src/main/java/edu/berkeley/boinc/PrefsFragment.java]: Please remove the additional indentation (one level).

Comment (adamradocz) [reply android/BOINC/app/src/main/java/edu/berkeley/boinc/PrefsFragment.java]: Done.

Comment (adamradocz) [reply android/BOINC/app/src/main/java/edu/berkeley/boinc/PrefsFragment.java]: Done

Comment (adamradocz) [reply android/BOINC/app/src/main/java/edu/berkeley/boinc/PrefsFragment.java]: Now should be fine.

**Comment by brevilo** [comment]
I presume you tested your latest implementation?

**Comment by brevilo** [comment]
Hm, any idea where or why that happens? While technically not a huge problem this is of course confusing to the user. So, if feasible, this should be fixed first.

**Comment by adamradocz** [comment]
Unfortunately, I've no idea.

**Comment by AenBleidd** [comment]
@you need to specify additional `PrefsListItemWrapperValue.Unit` type because default is integer, see:
https://github.com/BOINC/boinc/blob/9e18843a8dc69b6edc8da287ed57f1f9e5acfdf9/android/BOINC/app/src/main/java/edu/berkeley/boinc/adapter/PrefsListAdapter.java#L91-L93

**Comment by adamradocz** [comment]
Thanks @AenBleidd.
@brevilo It's done.

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/adapter/PrefsListAdapter.java [hunk 2]
```diff
@@ -106,6 +108,9 @@ public View getView(int position, View convertView, ViewGroup parent) {
 	    				case GIGABYTES:
 	    					value = Formatter.formatShortFileSize(this.activity, (long)(item.status.doubleValue() * 0x40000000));
 	    					break;
+						case DAYS:
```
Comment (AenBleidd) [inline android/BOINC/app/src/main/java/edu/berkeley/boinc/adapter/PrefsListAdapter.java]: Please fix indents here.
BTW, maybe it is better to add `UNIT` type `DECIMAL `as more generic than `DAYS`?

Comment (adamradocz) [reply android/BOINC/app/src/main/java/edu/berkeley/boinc/adapter/PrefsListAdapter.java]: You're right, changed and fixed.

**Comment by brevilo** [comment]
👍

**Pull request merged**